libraries/standard/utilities/include/iot_logging_setup.h

```diff
@@ -0,0 +1,272 @@
+        #define IotLogInfoWithArgs( pFormat, ... )     IotLog( IOT_LOG_INFO, pFormat, __VA_ARGS__ )
+        #define IotLogDebug( pFormat )                 IotLog( IOT_LOG_DEBUG, "%s", pFormat )
+        #define IotLogDebugWithArgs( pFormat, ... )    IotLog( IOT_LOG_DEBUG, pFormat, __VA_ARGS__ )
+
+/* Remove references to IotLog from the source code if logging is disabled. */
+    #elif LIBRARY_LOG_LEVEL == IOT_LOG_INFO
+        /* Debug messages will not be logged. All other macros will call IotLog */
+        #define IotLogError( pFormat )                 IotLog( IOT_LOG_ERROR, "%s", pFormat )
+        #define IotLogErrorWithArgs( pFormat, ... )    IotLog( IOT_LOG_ERROR, pFormat, __VA_ARGS__ )
+        #define IotLogWarn( pFormat )                  IotLog( IOT_LOG_WARN, "%s", pFormat )
+        #define IotLogWarnWithArgs( pFormat, ... )     IotLog( IOT_LOG_WARN, pFormat, __VA_ARGS__ )
+        #define IotLogInfo( pFormat )                  IotLog( IOT_LOG_INFO, "%s", pFormat )
+        #define IotLogInfoWithArgs( pFormat, ... )     IotLog( IOT_LOG_INFO, pFormat, __VA_ARGS__ )
+        #define IotLogDebug( pFormat )
+        #define IotLogDebugWithArgs( pFormat, ... )
+
+    #elif LIBRARY_LOG_LEVEL == IOT_LOG_WARN
+        /* Only "Warning" and IOT_LOG_ERROR messages will be logged.*/
+        #define IotLogError( pFormat )                 IotLog( IOT_LOG_ERROR, "%s", pFormat )
+        #define IotLogErrorWithArgs( pFormat, ... )    IotLog( IOT_LOG_ERROR, pFormat, __VA_ARGS__ )
+        #define IotLogWarn( pFormat )                  IotLog( IOT_LOG_WARN, "%s", pFormat )
+        #define IotLogWarnWithArgs( pFormat, ... )     IotLog( IOT_LOG_WARN, pFormat, __VA_ARGS__ )
+        #define IotLogInfo( pFormat )
+        #define IotLogInfoWithArgs( pFormat, ... )
+        #define IotLogDebug( pFormat )
+        #define IotLogDebugWithArgs( pFormat, ... )
+
+    #elif LIBRARY_LOG_LEVEL == IOT_LOG_ERROR
+        /* Only IOT_LOG_ERROR messages will be logged. */
+        #define IotLogError( pFormat )                 IotLog( IOT_LOG_ERROR, "%s", pFormat )
+        #define IotLogErrorWithArgs( pFormat, ... )    IotLog( IOT_LOG_ERROR, pFormat, __VA_ARGS__ )
+        #define IotLogWarn( pFormat )
+        #define IotLogWarnWithArgs( pFormat, ... )
+        #define IotLogInfo( pFormat )
+        #define IotLogInfoWithArgs( pFormat, ... )
+        #define IotLogDebug( pFormat )
+        #define IotLogDebugWithArgs( pFormat, ... )
+
+    #else /* if LIBRARY_LOG_LEVEL >= IOT_LOG_DEBUG */
+        #define IotLogError( ... )
+        #define IotLogWarn( ... )
+        #define IotLogInfo( ... )
+        #define IotLogDebug( ... )
+    #endif /* if LIBRARY_LOG_LEVEL >= IOT_LOG_DEBUG */
+#endif /* if !defined( LIBRARY_LOG_LEVEL ) || ( ( LIBRARY_LOG_LEVEL != IOT_LOG_NONE ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_ERROR ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_WARN ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_INFO ) && ( LIBRARY_LOG_LEVEL != IOT_LOG_DEBUG ) ) */
```

Comment:
Please remove this lengthy comment too.

Reply:
Re-ran uncrustify (after deleting the comment) to notice that it automatically adds the comment. Thus, will leave it here.